index.c

```diff
@@ -2285,18 +2285,28 @@ int mutt_index_menu(struct MuttWindow *dlg)
 #ifdef USE_NOTMUCH
       case OP_MAIN_VFOLDER_FROM_QUERY:
       case OP_MAIN_VFOLDER_FROM_QUERY_READONLY:
+      {
         buf[0] = '\0';
         if ((mutt_get_field("Query: ", buf, sizeof(buf), MUTT_NM_QUERY) != 0) || !buf[0])
         {
           mutt_message(_("No query, aborting"));
           break;
         }
-        if (!nm_uri_from_query(NULL, buf, sizeof(buf)))
-          mutt_message(_("Failed to create query, aborting"));
+
+        char *query_unencoded = mutt_str_strdup(buf);
+
+        if (nm_uri_from_query(NULL, buf, sizeof(buf)))
+        {
+          struct Mailbox *temp_mail = mx_path_resolve(buf);
+          temp_mail->name = mutt_str_strdup(query_unencoded);
```

Comment:
We could make this a little more efficient:

```suggestion
          temp_mail->name = query_unencoded;
          query_unencoded = NULL;
```